src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20220929133723_EES3369_AddContainedDatabaseUsers.sql

```diff
@@ -0,0 +1,128 @@
+    GRANT ALL ON [dbo].[ContentSections] TO [importer];
+    GRANT ALL ON [dbo].[DataImports] TO [importer];
+    GRANT ALL ON [dbo].[DataImportErrors] TO [importer];
+    GRANT ALL ON [dbo].[ExternalMethodology] TO [importer];
+    GRANT ALL ON [dbo].[Files] TO [importer];
+    GRANT ALL ON [dbo].[GlossaryEntries] TO [importer];
+    GRANT ALL ON [dbo].[LegacyReleases] TO [importer];
+    GRANT ALL ON [dbo].[Methodologies] TO [importer];
+    GRANT ALL ON [dbo].[MethodologyFiles] TO [importer];
+    GRANT ALL ON [dbo].[MethodologyNotes] TO [importer];
+    GRANT ALL ON [dbo].[MethodologyVersions] TO [importer];
+    GRANT ALL ON [dbo].[PublicationMethodologies] TO [importer];
+    GRANT ALL ON [dbo].[Publications] TO [importer];
+    GRANT ALL ON [dbo].[ReleaseContentBlocks] TO [importer];
+    GRANT ALL ON [dbo].[ReleaseContentSections] TO [importer];
+    GRANT ALL ON [dbo].[ReleaseFiles] TO [importer];
+    GRANT ALL ON [dbo].[Releases] TO [importer];
+    GRANT ALL ON [dbo].[ReleaseStatus] TO [importer];
+    GRANT ALL ON [dbo].[Themes] TO [importer];
+    GRANT ALL ON [dbo].[Topics] TO [importer];
+    GRANT ALL ON [dbo].[Update] TO [importer];
+End
+
+IF NOT EXISTS (SELECT name
+                FROM [sys].[database_principals]
+                WHERE name = 'publisher')
+Begin
+    CREATE USER [publisher] FROM LOGIN [publisher];
+
+    GRANT ALL ON [dbo].[Comment] TO [publisher];
+    GRANT ALL ON [dbo].[Contacts] TO [publisher];
+    GRANT ALL ON [dbo].[ContentBlock] TO [publisher];
+    GRANT ALL ON [dbo].[ContentSections] TO [publisher];
+    GRANT ALL ON [dbo].[DataImports] TO [publisher];
+    GRANT ALL ON [dbo].[DataImportErrors] TO [publisher];
+    GRANT ALL ON [dbo].[ExternalMethodology] TO [publisher];
+    GRANT ALL ON [dbo].[Files] TO [publisher];
+    GRANT ALL ON [dbo].[GlossaryEntries] TO [publisher];
+    GRANT ALL ON [dbo].[LegacyReleases] TO [publisher];
+    GRANT ALL ON [dbo].[Methodologies] TO [publisher];
+    GRANT ALL ON [dbo].[MethodologyFiles] TO [publisher];
+    GRANT ALL ON [dbo].[MethodologyNotes] TO [publisher];
+    GRANT ALL ON [dbo].[MethodologyVersions] TO [publisher];
+    GRANT ALL ON [dbo].[PublicationMethodologies] TO [publisher];
+    GRANT ALL ON [dbo].[Publications] TO [publisher];
+    GRANT ALL ON [dbo].[ReleaseContentBlocks] TO [publisher];
+    GRANT ALL ON [dbo].[ReleaseContentSections] TO [publisher];
+    GRANT ALL ON [dbo].[ReleaseFiles] TO [publisher];
+    GRANT ALL ON [dbo].[Releases] TO [publisher];
+    GRANT ALL ON [dbo].[ReleaseStatus] TO [publisher];
+    GRANT ALL ON [dbo].[Themes] TO [publisher];
+    GRANT ALL ON [dbo].[Topics] TO [publisher];
+    GRANT ALL ON [dbo].[Update] TO [publisher];
+end
+
+
+IF NOT EXISTS (SELECT name
+                FROM [sys].[database_principals]
+                WHERE name = 'content')
+Begin
+    CREATE USER [content] FROM LOGIN [content];
+    
+    GRANT SELECT ON [dbo].[Comment] TO [content];
```

Comment:
In the Azure environments we don't have a `content` or `data` user in the `content` database. We have an `adminapp` used by the Admin app service and a `publicapp` used by the Content and Data API app services.

In `appsettings.Development.json` in the Admin project I see the `ContentDb` connection string user is changing from SA to `adminapp` which I understand, but don't see a `CREATE USER` here for `adminapp` or any grants for it.

I'd also expect to either see a `CREATE USER` here for `publicapp` and grants for that, or some kind of commentary to say we are going to remove the `publicapp` user and switch to using new `content` and `data users`?

The description of this PR under the heading of "Content DB" says:

> No contained users will be created or permissions granted, as all the contained users already exist in that database. No permissions will be changed for these existing users by virtue of the database migrations only setting up permissions for NEW contained users.

I don't think that's correct because `content` and `data` don't exist, so they will be created in the Azure environments.

Reply:
Yes you're absolutely right!  I think for consistency then we should aim to use "content" and "data" users in the content db for the Content API and Data API, in the same way as we do for the public-statistics database at the moment.  I will add some instructions to the PR to decommission the `publicapp` user.

Reply:
Great, I think the text of the PR description under the headings "New contained users" -> "Content DB" is still out of date after the latest changes as it still says "No contained users will be created or permissions granted".

Reply:
Yup you're right!  I've updated it to:

```
### Content DB
The `data` and `content` contained users will be created in this environment and their permissions set up, by virtue of them not yet existing there (they exist in the `public-statistics` database currently but not the private stats or content dbs).  The `importer` and `publisher` users will remain untouched as they already exist here.
```

Thanks!